inference-engine/src/transformations/src/transformations/op_conversions/convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,372 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_padded2valid_conv.hpp"
+
+#include <memory>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/builder/reshape.hpp>
+#include <ngraph/builder/split.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+using namespace ngraph;
+using namespace op;
+
+bool IsTransposeOrderMatches(std::shared_ptr<Transpose> transpose, std::vector<int64_t> order) {
+    if (!transpose)
+        return false;
+    const Output<Node>& transpose_order = transpose->input_value(1);
+    auto transpose_order_dim = transpose_order.get_shape().size();
+
+    if (transpose_order_dim != 1 || transpose_order.get_shape()[0] != order.size())
+        return false;
+
+    auto const_with_order_values = std::dynamic_pointer_cast<ngraph::opset1::Constant>(transpose_order.get_node_shared_ptr());
+    if (!const_with_order_values)
+        return false;
+
+    const int64_t* data = const_with_order_values->get_data_ptr<int64_t>();
+    if (!data)
+        return false;
+
+    for (size_t i = 0; i < order.size(); i++) {
+        if (order[i] != data[i])
+            return false;
+    }
+
+    return true;
+}
+
+std::shared_ptr<opset1::StridedSlice> FlatCrop(Output<Node> input, size_t offset, size_t size) {
+    auto shape = input.get_shape();
+    if (shape.size() == 1) {
+        return std::make_shared<ngraph::opset1::StridedSlice>(
+            input, // data
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 1 }, { offset }), // begin slice index
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 1 }, { offset + size }), // end slice index
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 1 }, { 1 }), // strides
+            std::vector<int64_t>{0},  // begin mask
+            std::vector<int64_t>{0}); // end mask
+    } else if (shape.size() == 2) {
+        return std::make_shared<ngraph::opset1::StridedSlice>(
+            input, // data
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 2 }, { (size_t)0, offset }), // begin sice index
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 2 }, { (size_t)0, offset + size }), // end slice index
+            ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 2 }, { (size_t)1, (size_t)1 }), // strides
+            std::vector<int64_t>{1, 0},  // begin mask
+            std::vector<int64_t>{1, 0}); // end mask
+    }
+    return nullptr;
+}
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertPadded2ValidConv, "ConvertPadded2ValidConv", 0);
+bool ngraph::pass::ConvertPadded2ValidConv::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    // Traverse nGraph Function in topological order
```

Comment:
I'd split this function code into some separate functions, since it's too long and hard readable.

Reply:
I agree This function has over 300 LOC,
IMO single function should not be longer than 50 LOC.
However, IMO we can plan such refactoring to later, as this functionality is crucial.
I would recommend to create separate issue.

Reply:
Expect even worse with conv 2d decomposition... If you agree I'd like to put refactoring off until after the merge and do it together with fixing failing tests.

Reply:
ok